### PR TITLE
feat: allow usage of cached `gasUsage` + `SpeedProvider`

### PR DESCRIPTION
**Pull request by dohaki**
This PR enables two performance tweaks in the API:
- Using a cached `gasUnits` value for a fill - this saves an RPC request in the fee calculator
- Using the `SpeedProvider` which makes use of `Promise.any`

### CODE REVIEW

File: src/utils/common.ts [hunk 1]
```diff
@@ -262,11 +264,13 @@ export async function estimateTotalGasRequiredByUnsignedTransaction(
...
-    gasPrice,
-    gasLimit,
-  });
+  let nativeGasCost = gasUnits
+    ? BigNumber.from(gasUnits)
+    : await voidSigner.estimateGas({
+        ...unsignedTx,
+        gasPrice,
+        gasLimit,
+      });
```
Comment (pxrl) [inline src/utils/common.ts:273]: ooc, do you think it's a hard requirement to be able to specify `gasLimit` and `gasUnits` independently?

(I think the historical reason for needing to override gasLimit was probably debunked and we fortunately haven't hit another issue...at least as far as I am aware)

Comment (dohaki) [reply src/utils/common.ts:273]: Yea, that's a good point. We don't use that anywhere. I can remove that parameter then

Comment (dohaki) [reply src/utils/common.ts]: https://github.com/across-protocol/sdk/pull/717/commits/e662f595ec43410d79cf4c81be8414e6fa474cd5

**Review by gsteenkamp89 — APPROVED** [review]
Looks great! 🚀

**Review by james-a-morris — APPROVED** [review]
Is this a migration?

**Comment by dohaki** [comment]
> Is this a migration?

Can you elaborate a bit? Migration from what?

The context of these features introduced here are primarily for improving the performance of the API

**Pull request merged**